portfolio/src/main/webapp/script.js

```diff
@@ -105,3 +134,38 @@ function createCommentNode(comment) {
 
   return comDiv;
 }
+
+function onSignIn(googleUser) {
+  const profile = googleUser.getBasicProfile();
+  const id_token = googleUser.getAuthResponse().id_token;
+  
+  user = { 
+          name: profile.getName(),
+          imageUrl: profile.getImageUrl(),
+          email: profile.getEmail(),
+          tokenId: id_token,
+  }
+  toggleCommentForm();
+}
+
+function toggleCommentForm() {
+  const commentForm = document.getElementById("form");
+  const signinInfo = document.getElementById("signin-info");
+
+  if (user == null) {
+    commentForm.style.display = "none";
+    signinInfo.style.display = "block";
+  } else {
+    commentForm.style.display = "block";
+    signinInfo.style.display = "none";
+  }
+}
+
+function signOut() {
+  user = null;
+
+  let auth2 = gapi.auth2.getAuthInstance();
+  auth2.signOut().then(function () {
```

Comment:
I am pretty sure that we can do more things than a log line when signing out a user.

Reply:
I've also added a sign-out button :)